### PR TITLE
lavfi/dnn: Add OpenVINO API 2.0 support

### PR DESCRIPTION
**Pull request by wenbinc-Bin**
OpenVINO API 2.0 was released in March 2022, which introduced new features.
This commit implements current OpenVINO features with new 2.0 APIs. And will add other features in API 2.0.
Please add installation path, which include openvino.pc, to PKG_CONFIG_PATH mannually for new OpenVINO libs config.

### CODE REVIEW

File: libavfilter/dnn/dnn_backend_openvino.c [hunk 1]
```diff
@@ -51,9 +55,20 @@ typedef struct OVContext {
 typedef struct OVModel{
     OVContext ctx;
     DNNModel *model;
+#if HAVE_OPENVINO2
```
Comment (guoyejun) [inline libavfilter/dnn/dnn_backend_openvino.c:58]: looks that we support both ov2 and ov1?

Comment (wenbinc-Bin) [reply libavfilter/dnn/dnn_backend_openvino.c:58]: We support openvino. If it has openvino2 API, then we use openvino2 API.

Comment (xhaihao) [reply libavfilter/dnn/dnn_backend_openvino.c:58]: Will it work if there is only ov1 in the runtime but user builds FFmpeg against ov2 or inverse ?

Comment (wenbinc-Bin) [reply libavfilter/dnn/dnn_backend_openvino.c:58]: No it won't work. We need to make sure it uses same env when both compiling and running.

**Comment by wenbinc-Bin** [comment]
https://patchwork.ffmpeg.org/project/ffmpeg/patch/20230815082631.2648570-1-wenbin.chen@intel.com/

**Pull request closed without merging**